BGL/include/CGAL/boost/graph/named_params_helper.h

```diff
@@ -112,6 +114,205 @@ namespace CGAL {
+  {
+    CGAL::helpers::init_face_indices(g, map);
+  }
+  template< class PMap, class Graph, typename T>
+  void initialize_index_map(PMap map, const Graph& g, const CGAL::dynamic_face_property_t<T>&, const CGAL::Tag_true&)
+  {
+    CGAL::helpers::init_face_indices(g, map);
+  }
+  
+  //default case : don't do anything
+  template<typename SimplexTag, class PMap, class Graph, typename IsWritableTag>
+  void initialize_index_map(PMap, const Graph&, const SimplexTag& t, const IsWritableTag& w)
+  {}
+
+  
+#define CGAL_IS_PMAP_WRITABLE(TAG) template<typename IsRefConst> \
+  struct Is_pmap_writable<TAG, IsRefConst>{ typedef CGAL::Tag_true result; }; \
+  template<> \
+  struct Is_pmap_writable<TAG,std::integral_constant<bool, true> >{ typedef CGAL::Tag_false result; };
+
+  //Default is false
+  template<typename PMapCategory, typename IsRefConst>
+  struct Is_pmap_writable{
+    typedef CGAL::Tag_false result;
+  };
+
+  //Pmaps with these tags will be considered writable, unless their reference is const
+  CGAL_IS_PMAP_WRITABLE(boost::read_write_property_map_tag)
+  CGAL_IS_PMAP_WRITABLE(boost::writable_property_map_tag)
+  CGAL_IS_PMAP_WRITABLE(boost::lvalue_property_map_tag)
+#undef CGAL_IS_PMAP_WRITABLE
+
+  //overloads used to select a default map:
+  // use the one passed in the named parameters (user must have initialized it)
+  template <class MapFromNP, class Default_tag, class Dynamic_tag, class Mesh>
+  MapFromNP
+  get_ndi_map(MapFromNP m, Default_tag, Dynamic_tag, const Mesh&)
+  {
+    return m;
+  }
+
+  // use the one internal to the mesh (it will be init if writable)
+  template <class Default_tag, class Dynamic_tag, class Mesh>
+  typename boost::property_map<Mesh, Default_tag >::const_type
+  get_ndi_map(CGAL::internal_np::Param_not_found, Default_tag t, Dynamic_tag , const Mesh& m)
+  {
+    typedef typename boost::property_map<Mesh, Default_tag >::const_type Map_const_type;
+    typedef typename boost::property_traits
+        <typename boost::property_map<Mesh, Default_tag >
+        ::const_type>::category Category;
+    typedef typename boost::property_traits<typename boost::property_map<Mesh, Default_tag >
+        ::const_type>::reference Reference;
+    
+    Map_const_type map = get(t, m);
+    typename Is_pmap_writable<Category,
+        std::is_const<Reference> >::result is_writable;
+    initialize_index_map(map, m, t,is_writable);
+    return map;
+  }
+
+  // create a dynamic property and initialize it
+  template <class Dynamic_tag, class Mesh>
+  typename boost::property_map<Mesh, Dynamic_tag >::const_type
+  get_ndi_map(CGAL::internal_np::Param_not_found, Dynamic_tag t, Dynamic_tag , const Mesh& m)
+  {
+    typedef typename boost::property_map<Mesh, Dynamic_tag >::const_type Map_const_type;
+    Map_const_type map = get(t,m);
+    initialize_index_map(map, m, t, CGAL::Tag_true());
+    return map;
+  }
+
+  }//end of internal_np
+
+  namespace Polygon_mesh_processing
```

Comment:
Why is this in PMP and not directly in CGAL? Seems like it should be useful at plenty of other places too, no?